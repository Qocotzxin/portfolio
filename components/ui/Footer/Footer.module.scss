.Footer {
  width: 100%;
  height: 4rem;
  padding-left: var(--chakra-space-8);
  padding-right: var(--chakra-space-8);
  background: white;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  bottom: 0;
  left: 0;

  &--dark {
    background: var(--chakra-colors-gray-800);
  }

  &__block {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
