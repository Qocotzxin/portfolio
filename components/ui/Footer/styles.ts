import styles from "./Footer.module.scss";

export const Classes = {
  base: styles.Footer,
  baseDark: styles["Footer--dark"],
  block: styles.Footer__block,
};
