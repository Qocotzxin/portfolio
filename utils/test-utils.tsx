import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
