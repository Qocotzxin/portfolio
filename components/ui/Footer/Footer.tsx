import { useIsDarkMode } from "@hooks/useIsDarkMode";
import FooterLinks from "@ui/Footer/FooterLinks";
import LanguageSelector from "@ui/Footer/LanguageSelector";
import { applyConditionalClass, concatClassNames } from "@utils/classnames";
import { FC } from "react";
import { Classes } from "./styles";

const Footer: FC = () => {
  const isDark = useIsDarkMode();

  return (
    <div
      className={concatClassNames(
        Classes.base,
        applyConditionalClass(isDark, Classes.baseDark)
      )}
    >
      <FooterLinks />
      <div className={Classes.block}>
        <LanguageSelector />
      </div>
    </div>
  );
};

export default Footer;
