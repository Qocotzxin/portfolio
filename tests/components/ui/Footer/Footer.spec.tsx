import Footer from "@ui/Footer";
import { render, screen } from "@utils/test-utils";

describe("Footer.", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("Should render the language selector.", () => {
    render(<Footer />);
    expect(screen.getByTestId("LanguageSelector")).toBeInTheDocument();
  });
});
