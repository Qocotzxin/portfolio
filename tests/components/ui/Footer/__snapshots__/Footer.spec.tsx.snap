// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer. Should match snapshot. 1`] = `
<div>
  <div
    class="Footer Footer--dark"
  >
    <div
      class="FooterLinks"
    >
      <a
        aria-label="Visit Github"
        href="https://github.com"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          class="FooterLinks__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
          />
        </svg>
      </a>
    </div>
    <div
      class="Footer__block"
    >
      <div
        class="FloatingMenu"
        data-testid="FloatingMenu"
      >
        <button
          class="Button Button--dark"
          data-testid="LanguageSelector"
          id="FloatingMenu-trigger"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95a15.65 15.65 0 00-1.38-3.56A8.03 8.03 0 0118.92 8zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56A7.987 7.987 0 015.08 16zm2.95-8H5.08a7.987 7.987 0 014.33-3.56A15.65 15.65 0 008.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95a8.03 8.03 0 01-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
        </button>
        <div
          aria-labelledby="FloatingMenu-trigger"
          class="FloatingMenu__list  FloatingMenu__list--dark"
          role="menu"
          tabindex="-1"
        >
          <a
            aria-current="false"
            class="Link Link--dark"
            data-order="0"
            href="/#main"
            role="menuitem"
            style="order: 0;"
            tabindex="-1"
          >
            english
          </a>
          <a
            aria-current="false"
            class="Link Link--dark"
            data-order="1"
            href="/#main"
            role="menuitem"
            style="order: 1;"
            tabindex="-1"
          >
            spanish
          </a>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
